layers/+tools/lsp/funcs.el

```diff
@@ -144,4 +165,55 @@ https://github.com/emacs-lsp/lsp-javascript/issues/9#issuecomment-379515379"
               (push `((,point0 . ,point1) . ,w) candidates))))))
     (avy-with avy-document-symbol
       (avy--process candidates
-                    (avy--style-fn avy-style)))))
+        (avy--style-fn avy-style)))))
+
+
+;; These functions facilitate extension of the navigation-mode keybindings in derived layers
+;; See c/c++ layer for a usage example
+(defun spacemacs//lsp-get-extension-name (layer-name nav-mode kind)
+  (intern (concat layer-name "/" nav-mode "-" (symbol-name kind))))
+
+(defun spacemacs/lsp-get-find-extension-name (layer-name kind)
+  (spacemacs//lsp-get-extension-name layer-name "find" kind))
+
+(defun spacemacs/lsp-get-peek-extension-name (layer-name kind)
+  (spacemacs//lsp-get-extension-name layer-name "peek" kind))
+
+(defun spacemacs//lsp-define-find-extension (layer-name kind request &optional extra)
+  (if extra
+    (defalias (spacemacs/lsp-get-find-extension-name layer-name kind) `(lambda () ,(format "Find %s" (symbol-name kind)) (interactive) (lsp-find-custom ,request ',extra)))
+    (defalias (spacemacs/lsp-get-find-extension-name layer-name kind) `(lambda () ,(format "Find %s" (symbol-name kind)) (interactive) (lsp-find-custom ,request)))))
+
+(defun spacemacs//lsp-define-peek-extension (layer-name kind request &optional extra)
+  (if extra
+    (defalias (spacemacs/lsp-get-peek-extension-name layer-name kind) `(lambda () ,(format "Peek %s" (symbol-name kind)) (interactive) (lsp-ui-peek-find-custom ',kind ,request ',extra)))
```

Comment:
Heads-up. The extra `kind` parameter in `lsp-ui-peek-*` has been removed.

`lsp-find-custom` `lsp-ui-peek-find-custom` can be used the same way now.

Reply:
Thanks for the heads-up Fangrui!
I've updated, amended, pushed. Nice that `lsp-find-custom` and `lsp-ui-peek-find-custom` have the same call syntax now. It's allowed me to simplify the chunk of elisp that defines the extensions considerably.